plugins/modules/net_tools/ldap/ldap_search.py

```diff
@@ -106,11 +106,18 @@ def main():
         module.fail_json(msg=missing_required_lib('python-ldap'),
                          exception=LDAP_IMP_ERR)
 
-    if not module.check_mode:
-        try:
-            LdapSearch(module).main()
-        except Exception as exception:
-            module.fail_json(msg="Attribute action failed.", details=to_native(exception))
+    # remove the following and allow to run even in check mode #3619
+    # if not module.check_mode:
+    #     try:
+    #         LdapSearch(module).main()
+    #     except Exception as exception:
+    #         module.fail_json(msg="Attribute action failed.", details=to_native(exception))
+
+    # Run even in check mode #3619
```

Comment:
No need to keep that commented - just remove it. If we ever need to recall what it was before, that's what git is here for ;-)